runbot_gitlab/controllers/gitlab_ci_controller.py

```diff
@@ -39,11 +39,26 @@ def hook_gitlab(self, repo_id=None, **post):
                 return self.hook(repo.id, **post)
             if (event == "build" and data["build_name"] in ("build_deployv", "build_docker") and
                 data["build_status"] == "success" and repo and repo.is_t2d_deployv):
-                build_domain = [("repo_id", "=", repo.id), ("name", "=", data["sha"])]
-                build = request.env["runbot.build"].sudo().search(build_domain, order='id DESC', limit=1)
-                if build:
-                    build.write({"deployv_image_built": True})
-                    forced_builds = build._force("Image built so rebuild runbot job")
+                build = request.env["runbot.build"].sudo()
+                query = """SELECT id FROM runbot_build WHERE id IN (
```

Comment:
You don't need a subquery here, only using WHERE and HAVING clauses is enough

Reply:
I thought the same but was not working well for this purpose

Try it directly in server action and you will see it

Reply:
My first try was using read_group with max:id

But it is not working for Odoo v11.0

Then I used the way than you said but it requires that the field exists in the SELECT in order to be able to be filtered in the HAVING

Then I used the inefficient way (the current one) but generating indexes and it is working well

The analyze results was MS

```sql
EXPLAIN (ANALYZE, VERBOSE, BUFFERS)
SELECT id FROM runbot_build WHERE id IN (
    SELECT MAX(id) AS build_id
    FROM runbot_build
    WHERE repo_id IN (231)
        AND name = 'd42e3ec27d8e8eadc7b4d4aa67268dc6ec37fc8e'
    GROUP BY branch_id)
AND state='done' AND result='skipped' AND deployv_image_built IS NOT TRUE
```


```txt
 Planning Time: 0.561 ms
 Execution Time: 0.321 ms
```

So, I stopped to improving here the query

Of course I could use a aggregator method with order by

But really I decided stop over-improving here